### DISCUSSION TITLE
How to execute onnxruntime benchmarks through Migraphx ?

### DISCUSSION THREAD

**Discussion by jasberc** [ROCm/AMDMIGraphX#2359] [Q&A] [answered]
I would like to execute some BERT related benchmarks ex:
python3 -m onnxruntime.transformers.benchmark -g -m bert-base-cased --provider migraphx
But I got error as follows.
RuntimeError: Found no NVIDIA driver on your system.

However I confirm I have required execution providers are installed.
>>> import onnxruntime
>>> onnxruntime.get_available_providers()
['MIGraphXExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']

anyone knows what's going on ? thanks.

**Comment by causten** [comment]
You need to build ONNX Runtime with ROCM and MIGraphX enabled.   See https://github.com/microsoft/onnxruntime/blob/main/dockerfiles/Dockerfile.migraphx#L56 for the build instructions

**Reply by causten** [reply, marked as answer]
Installing Torch without ROCm can also cause this. i.e. if you ran `pip install torch`. That version of torch is fully mapped to cuda.  You have two options uninstall torch and install via a rocm repo `pip install torch==2.0.1 --trusted-host https://repo.radeon.com/ -f https://repo.radeon.com/rocm/manylinux/rocm-rel-5.7/`  or use the rocm pytorch docker and install ORT in it... `rocm/pytorch:latest`

**Reply by jasberc** [reply]
Thanks for your replying. My problem is fixed by reinstalling a version of torch with ROCm.
